core/src/test/scala/dagr/core/config/ConfigurationTest.scala

```diff
@@ -102,9 +105,24 @@ class ConfigurationTest extends UnitSpec {
 
   it should "find executables" in {
     conf.configureExecutable("some-executable", "n/a") shouldBe Paths.get("/does/not/exist")
+    conf.configureExecutableFromBinDirectory("some-executable", "exec") shouldBe Paths.get("/does/not/exist/exec")
 
-    val java = conf.configureExecutable("java.exe", "java")
+    var java = conf.configureExecutable("java.exe", "java")
     java.getFileName.toString shouldBe "java"
     Files.isExecutable(java) shouldBe true
+    java = conf.configureExecutableFromBinDirectory("path-does-not-exist", "java")
+    java.getFileName.toString shouldBe "java"
+    Files.isExecutable(java) shouldBe true
+    Configuration.requestedKeys should contain ("java.exe")
+  }
+
+  it should "thrown an exception when an executable cannot be found" in {
+    an[Generic] should be thrownBy conf.configureExecutable("some-executable-not-found", "n/a")
```

Comment:
I would have this check for ConfigException, so that if we change our minds about the right subtype later the tests don't break.